### PR TITLE
[cli] [tabletmanager] Migrate all tabletmanager flags to `pflag`

### LINKED ISSUE #10980: Switch flag definitions to be on pflag instead of flag in `package go/vt/vttablet/tabletmanager`
Part of https://github.com/vitessio/vitess/issues/10697.

Current flags:

```
$ git grep -E "\bflag\.[A-Z]" -- go/vt/vttablet/tabletmanager/*.go
go/vt/vttablet/tabletmanager/orchestrator.go:   orcAddr     = flag.String("orc_api_url", "", "Address of Orchestrator's HTTP API (e.g. http://host:port/api/). Leave empty to disable Orchestrator integration.")
go/vt/vttablet/tabletmanager/orchestrator.go:   orcUser     = flag.String("orc_api_user", "", "(Optional) Basic auth username to authenticate with Orchestrator's HTTP API. Leave empty to disable basic auth.")
go/vt/vttablet/tabletmanager/orchestrator.go:   orcPassword = flag.String("orc_api_password", "", "(Optional) Basic auth password to authenticate with Orchestrator's HTTP API.")
go/vt/vttablet/tabletmanager/orchestrator.go:   orcTimeout  = flag.Duration("orc_timeout", 30*time.Second, "Timeout for calls to Orchestrator's HTTP API")
go/vt/vttablet/tabletmanager/orchestrator.go:   orcInterval = flag.Duration("orc_discover_interval", 0, "How often to ping Orchestrator's HTTP API endpoint to tell it we exist. 0 means never.")
go/vt/vttablet/tabletmanager/restore.go:        restoreFromBackup      = flag.Bool("restore_from_backup", false, "(init restore parameter) will check BackupStorage for a recent backup at startup and start there")
go/vt/vttablet/tabletmanager/restore.go:        restoreFromBackupTsStr = flag.String("restore_from_backup_ts", "", "(init restore parameter) if set, restore the latest backup taken at or before this timestamp. Example: '2021-04-29.133050'")
go/vt/vttablet/tabletmanager/restore.go:        restoreConcurrency     = flag.Int("restore_concurrency", 4, "(init restore parameter) how many concurrent files to restore at once")
go/vt/vttablet/tabletmanager/restore.go:        waitForBackupInterval  = flag.Duration("wait_for_backup_interval", 0, "(init restore parameter) if this is greater than 0, instead of starting up empty when no backups are found, keep checking at this interval for a backup to appear")
go/vt/vttablet/tabletmanager/restore.go:        binlogHost           = flag.String("binlog_host", "", "PITR restore parameter: hostname/IP of binlog server.")
go/vt/vttablet/tabletmanager/restore.go:        binlogPort           = flag.Int("binlog_port", 0, "PITR restore parameter: port of binlog server.")
go/vt/vttablet/tabletmanager/restore.go:        binlogUser           = flag.String("binlog_user", "", "PITR restore parameter: username of binlog server.")
go/vt/vttablet/tabletmanager/restore.go:        binlogPwd            = flag.String("binlog_password", "", "PITR restore parameter: password of binlog server.")
go/vt/vttablet/tabletmanager/restore.go:        timeoutForGTIDLookup = flag.Duration("pitr_gtid_lookup_timeout", 60*time.Second, "PITR restore parameter: timeout for fetching gtid from timestamp.")
go/vt/vttablet/tabletmanager/restore.go:        binlogSslCa          = flag.String("binlog_ssl_ca", "", "PITR restore parameter: Filename containing TLS CA certificate to verify binlog server TLS certificate against.")
go/vt/vttablet/tabletmanager/restore.go:        binlogSslCert        = flag.String("binlog_ssl_cert", "", "PITR restore parameter: Filename containing mTLS client certificate to present to binlog server as authentication.")
go/vt/vttablet/tabletmanager/restore.go:        binlogSslKey         = flag.String("binlog_ssl_key", "", "PITR restore parameter: Filename containing mTLS client private key for use in binlog server authentication.")
go/vt/vttablet/tabletmanager/restore.go:        binlogSslServerName  = flag.String("binlog_ssl_server_name", "", "PITR restore parameter: TLS server name (common name) to verify against for the binlog server we are connecting to (If not set: use the hostname or IP supplied in -binlog_host).")
go/vt/vttablet/tabletmanager/rpc_lock_tables.go:        lockTablesTimeout = flag.Duration("lock_tables_timeout", 1*time.Minute, "How long to keep the table locked before timing out")
go/vt/vttablet/tabletmanager/rpc_replication.go:        _                = flag.Bool("enable_semi_sync", false, "DEPRECATED - Set the correct durability policy on the keyspace instead.")
go/vt/vttablet/tabletmanager/rpc_replication.go:        setSuperReadOnly = flag.Bool("use_super_read_only", false, "Set super_read_only flag when performing planned failover.")
go/vt/vttablet/tabletmanager/shard_sync.go:     shardSyncRetryDelay = flag.Duration("shard_sync_retry_delay", 30*time.Second, "delay between retries of updates to keep the tablet and its shard record in sync")
go/vt/vttablet/tabletmanager/tm_init.go:        tabletHostname     = flag.String("tablet_hostname", "", "if not empty, this hostname will be assumed instead of trying to resolve it")
go/vt/vttablet/tabletmanager/tm_init.go:        initKeyspace       = flag.String("init_keyspace", "", "(init parameter) keyspace to use for this tablet")
go/vt/vttablet/tabletmanager/tm_init.go:        initShard          = flag.String("init_shard", "", "(init parameter) shard to use for this tablet")
go/vt/vttablet/tabletmanager/tm_init.go:        initTabletType     = flag.String("init_tablet_type", "", "(init parameter) the tablet type to use for this tablet.")
go/vt/vttablet/tabletmanager/tm_init.go:        initDbNameOverride = flag.String("init_db_name_override", "", "(init parameter) override the name of the db used by vttablet. Without this flag, the db name defaults to vt_<keyspacename>")
go/vt/vttablet/tabletmanager/tm_init.go:        skipBuildInfoTags  = flag.String("vttablet_skip_buildinfo_tags", "/.*/", "comma-separated list of buildinfo tags to skip from merging with -init_tags. each tag is either an exact match or a regular expression of the form '/regexp/'.")
go/vt/vttablet/tabletmanager/tm_init.go:        initPopulateMetadata = flag.Bool("init_populate_metadata", false, "(init parameter) populate metadata tables even if restore_from_backup is disabled. If restore_from_backup is enabled, metadata tables are always populated regardless of this flag.")
go/vt/vttablet/tabletmanager/tm_init.go:        initTimeout          = flag.Duration("init_timeout", 1*time.Minute, "(init parameter) timeout to use for the init phase.")
go/vt/vttablet/tabletmanager/tm_init.go:        flag.Var(&initTags, "init_tags", "(init parameter) comma separated list of key:value pairs used to tag the tablet")
go/vt/vttablet/tabletmanager/tm_state.go:var publishRetryInterval = flag.Duration("publish_retry_interval", 30*time.Second, "how long vttablet waits to retry publishing the tablet record")
```

### PR DESCRIPTION
**Pull request by ajm188**
## Description

Does what it says on the tin. This one was pretty straightforward.

## Related Issue(s)

Closes #10980.

## Checklist

-   [ ] "Backport me!" label has been added if this change should be backported
-   [x] Tests were added or are not required
-   [x] Documentation was added or is not required

### CODE REVIEW

File: doc/SeparatingVttabletMysql.md [hunk 1]
```diff
@@ -25,7 +25,7 @@ The following adjustments need to be made to VTTablet command line parameters:
   `-db\_socket` parameters.
 
 * Disable restores / backups, by not passing any backup command line
-  parameters. Specifically, `-restore_from_backup` and
+  parameters. Specifically, `--restore_from_backup` and
```
Comment (ajm188) [inline doc/SeparatingVttabletMysql.md:28]: I didn't want to explode this PR (also it's extremely tangential to this unit of work) by updating everything here, so I filed https://github.com/vitessio/vitess/issues/11058

File: go/flags/endtoend/vttablet.txt [hunk 2]
```diff
@@ -195,7 +195,6 @@ Usage of vttablet:
       --enable_lag_throttler                                             If true, vttablet will run a throttler service, and will implicitly enable heartbeats
       --enable_query_plan_field_caching                                  (DEPRECATED) This option fetches & caches fields (columns) when storing query plans (default true)
       --enable_replication_reporter                                      Use polling to track replication lag.
-      --enable_semi_sync                                                 DEPRECATED - Set the correct durability policy on the keyspace instead.
```
Comment (GuptaManan100) [inline go/flags/endtoend/vttablet.txt:198]: Deprecated flags don't show up in the output?

**Review by GuptaManan100 — COMMENTED** [review]
Everything else LGTM

**Review by GuptaManan100 — APPROVED** [review]

File: go/flags/endtoend/vttablet.txt (hunk 2 above)
Comment (ajm188) [reply go/flags/endtoend/vttablet.txt:198]: yep! instead, when you use them (which still works), the deprecation message (second arg to `MarkDeprecated`) is printed to the command's stderr `io.Writer` ([c.f.](https://pkg.go.dev/github.com/spf13/pflag#FlagSet.MarkDeprecated))

Comment (GuptaManan100) [reply go/flags/endtoend/vttablet.txt:198]: This is perfect!!

**Review by rsajwani — COMMENTED** [review]
LGTM

**Pull request merged**